### PR TITLE
Use Iss field instead of domain as a refresh URL 

### PR DESCRIPTION
**Pull request by MrFlashAccount**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by indiv0 — APPROVED** [review]
LGTM, will do QA now

File: app/ide-desktop/lib/dashboard/src/authentication/cognito.ts [hunk 1]
```diff
@@ -411,9 +403,29 @@ function parseUserSession(
...
+ * @throws Error if the `iss` field of the payload is not a valid URL.
+ */
+function extractRefreshUrlFromSession(session: cognito.CognitoUserSession): string {
+  const { iss } = session.getAccessToken().payload
+
+  if (typeof iss !== 'string') {
+    throw new Error('Payload does not have an iss field.')
+  } else {
+    try {
+      new URL(iss)
```
Comment (somebody1234) [inline app/ide-desktop/lib/dashboard/src/authentication/cognito.ts]: `return new URL(iss).href`

**Review by somebody1234 — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/authentication/cognito.ts (hunk 1 above)
Comment (radeusgd) [reply app/ide-desktop/lib/dashboard/src/authentication/cognito.ts]: btw. I just had a look at the linked doc and it says that the `iss` field has the form:
```
https://cognito-idp.<Region>.amazonaws.com/<your user pool ID>
```

Should our URL also include the pool ID or should that be stripped?

**Comment by indiv0** [comment]
QA passed ✅ 

image

File: app/ide-desktop/lib/dashboard/src/authentication/cognito.ts (hunk 1 above)
Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/authentication/cognito.ts]: Yea, on the second thought `return new URL(iss).toString()` looks even better

Comment (radeusgd) [reply app/ide-desktop/lib/dashboard/src/authentication/cognito.ts]: Okay, I've just did a practical test - I rebased on top of your PR and checked the new `refresh_url` - it does respond with a successfully refreshed token.

So this seems OK.

**Review by radeusgd — APPROVED** [review]
Thanks for such a quick fix

**Pull request merged**